SENR0001: Error description updated

ChristianGruen opened the pull request:
Observed by @line-o:
https://xmlcom.slack.com/archives/C01GVC3JLHE/p1689946671105499

line-o commented:
I would propose to reflect the change in https://qt4cg.org/specifications/xpath-functions-40/Overview.html#func-serialize under error conditions as well. It currently states:

> The serialization process will raise an error if $input is an attribute or namespace node.

And could be modified to 

> The serialization process will raise an error `SENR0001` if $input is an attribute, namespace node, function or map.

ChristianGruen commented:
I’ve dropped this rule as it is only valid for a subset of the serialization methods.
In addition, I’ve updated the type of the `$options` parameter.

line-o commented:
I agree on the options but the second paragraph you dropped is about $input

ChristianGruen commented:
The rule for `$input` was dropped as it is only valid for a subset of the serialization methods.

What do others think: Should we keep this specific serialization error in the XQFO spec and give more details on it?

ChristianGruen commented:
To be revised (https://github.com/qt4cg/qtspecs/commit/9a785062ac172f4fef41989f17f915339170e337#commitcomment-122390463)

line-o commented:
As an implementor I would like to see which _common_ errors are expected to be raised on typical conditions. I would therefore like to see the mention of serialization input error be generalised as such:

> The serialization process will raise an error SENR0001 if `$input` is of a type that cannot be serialised with the given method option.

ndw commented:
The CG agreed to merge this issue at [meeting 043](https://qt4cg.org/meeting/minutes/2023/07-25.html)
